modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp

```diff
@@ -51,6 +51,33 @@ std::string LemDCBM400600Controller::get_current_transaction() {
     }
 }
 
+void LemDCBM400600Controller::update_lem_status() {
+    // should call this after a communication error to figure out what has been happening
+    auto status_response = this->http_client->get("/v1/status");
+
+    if (status_response.status_code != 200) {
+        throw UnexpectedDCBMResponseCode("/v1/status", 200, status_response);
+    }
+    try {
+        json data = json::parse(status_response.body);
+        this->transaction_is_ongoing_at_startup = data.at("status").at("bits").at("transactionIsOnGoing");
+        this->current_transaction_id = get_current_transaction();
```

Comment:
this looks strange, why is the `current_transaction_id` fetched in this try block? Does it depend on whether the `transaction_is_ongoing_at_startup` can be read from the json?